planner/core/point_get_plan.go

```diff
@@ -602,6 +611,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 		p.Handle = intDatum.GetInt64()
 		p.UnsignedHandle = mysql.HasUnsignedFlag(fieldType.Flag)
 		p.HandleParam = handlePair.param
+		if pi != nil {
+			partitionSchema, colNames := buildSchemaForPartition(tblName.Schema, tbl, tblAlias)
+			exprs, err := expression.ParseSimpleExprsWithNames(ctx, pi.Expr, partitionSchema, colNames)
+			if err != nil {
+				return nil
+			}
+			pos, ok := locateHashPartition(exprs[0], pairs)
```

Comment:
Is it possible that `len(exprs) > 1`?

Reply:
no, expression will be checked when creating table.